### PR TITLE
Improved local development with Testcontainers, Localstack, Moto and better Gradle configs

### PR DESCRIPTION
**Pull request by styczynski**
This PR provides a fully functional project setup.
The following changes were made:

* Remove local initializations of Amazon SDK clients for sake of autowired config
* Make `frontend` a valid Gradle subproject
* Move `gradlew` and stuff into root folder of project
* Setup Testcontainers with Localstack, Moto and custom Docker image for budgets

# New building tasks:

## Development

To run a project in development mode please open two terminals and type in one of them the following command:
```
    $ ./gradlew startFrontend
```
Frontend has hot-reload features. You can restart the server without interrupting Gradle just by typing:
```
   rs
```

And in the other one:
```
   $ ./gradlew startBackend
```

The mentioned command will launch backend server on `localhost:8100` and frontend on `localhost:8080`.

## Running tests

To run tests (integration tests with full setup of database, fake AWS etc.) please type:
```
   $ ./gradlew integrationTest
```

## Creating Jar

The normal jar-creation process looks the same i.e:
```
    $ ./gradlew bootJar
```

### CODE REVIEW

File: frontend/configs/webpack/common.js [hunk 1]
```diff
@@ -38,11 +39,10 @@ module.exports = {
...
-                  fix: true
-              }
-           }
-        ]
-      },
+              },
+           },
+        ],
+      },*/
       {
```
Comment (Ogochi) [inline frontend/configs/webpack/common.js]: Why commented?

File: server/src/main/java/com/omigost/server/config/AWSLocalstackConfig.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    private static MotoContainer motoIAM = new MotoContainer(MotoContainer.Service.IAM);
+
+    private static MotoContainer motoOrganizations = new MotoContainer(MotoContainer.Service.ORGANIZATIONS);
+
+    private static MotoContainer motoEC2 = new MotoContainer(MotoContainer.Service.EC2);
+
+    private static BudgetsContainer budgetsContainer = new BudgetsContainer();
+
+    private static PostgreSQLContainer postgreSQLContainer = new PostgreSQLContainer("postgres:10.4");
```
Comment (Ogochi) [inline server/src/main/java/com/omigost/server/config/AWSLocalstackConfig.java]: PostgreSQL version should be somewhere else. maybe project config?

File: server/src/main/java/com/omigost/server/config/AWSConfig.java [hunk 3]
```diff
@@ -3,20 +3,87 @@
...
+import com.amazonaws.services.sns.AmazonSNSClientBuilder;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
 
 @Configuration
 @Slf4j
+@Profile("default")
```
Comment (Ogochi) [inline server/src/main/java/com/omigost/server/config/AWSConfig.java]: Why do we need it?

File: server/src/main/java/com/omigost/server/localstack/AWSServiceImageContainer.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    public AwsClientBuilder.EndpointConfiguration getEndpointConfiguration() {
+        final String address = getContainerIpAddress();
+        String ipAddress = address;
+        try {
+            ipAddress = InetAddress.getByName(address).getHostAddress();
+        } catch (UnknownHostException ignored) {
+
+        }
```
Comment (Ogochi) [inline server/src/main/java/com/omigost/server/localstack/AWSServiceImageContainer.java]: Add logging

File: server/src/main/java/com/omigost/server/localstack/AWSServiceImageContainer.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        withFileSystemBind("//var/run/docker.sock", "/var/run/docker.sock");
+        waitingFor(Wait.forLogMessage(".*" + getServiceStartMessage() + ".*", 1));
+    }
+
+    public abstract String getServiceStartMessage();
+
+    public abstract String getServiceImageName();
+
+    public abstract int getServicePort();
+
```
Comment (Ogochi) [inline server/src/main/java/com/omigost/server/localstack/AWSServiceImageContainer.java]: Do we need this abstract methods lines? Doesn't it work without them?

File: server/src/main/java/com/omigost/server/localstack/AWSServiceImageContainer.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            } catch (UnknownHostException ignored) {
+
+            }
+        }
+
+        return new AwsClientBuilder.EndpointConfiguration(
+                "http://" +
+                        ipAddress +
+                        ":" +
+                        getMappedPort(getServicePort()), "us-east-1");
```
Comment (Ogochi) [inline server/src/main/java/com/omigost/server/localstack/AWSServiceImageContainer.java]: We can use here region specified in config.

File: server/src/main/java/com/omigost/server/localstack/BudgetsContainer.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
+package com.omigost.server.localstack;
+
+public class BudgetsContainer extends AWSServiceImageContainer {
+
+    @Override
+    public String getServiceImageName() {
+        return "styczynski/local-aws-budgets";
```
Comment (Ogochi) [inline server/src/main/java/com/omigost/server/localstack/BudgetsContainer.java]: Please change to sth more neutral than "styczynski" xD

File: frontend/configs/webpack/common.js (hunk 1 above)
Comment (styczynski) [reply frontend/configs/webpack/common.js]: There were problems with linter now the linter runs as separate command `yarn run lint` .

File: server/src/main/java/com/omigost/server/config/AWSLocalstackConfig.java (hunk 2 above)
Comment (styczynski) [reply server/src/main/java/com/omigost/server/config/AWSLocalstackConfig.java]: That was fixed by moving all hardcoded image names and ports to the properties file.

**Pull request merged**